en/docs/chapter_searching/binary_search_insertion.md

```diff
@@ -32,21 +32,21 @@ Therefore, at the end of the binary, it is certain that: $i$ points to the first
 
     Based on the previous question, assume the array may contain duplicate elements, all else remains the same.
 
-Suppose there are multiple `target`s in the array, ordinary binary search can only return the index of one of the `target`s, **and it cannot determine how many `target`s are to the left and right of that element**.
+When there are multiple occurrences of `target` in the array, a regular binary search can only return the index of one occurrence of `target`, **and it cannot determine how many occurrences of `target` are to the left and right of that position**.
 
-The task requires inserting the target element to the very left, **so we need to find the index of the leftmost `target` in the array**. Initially consider implementing this through the steps shown in the figure below.
+The problem requires inserting the target element to the very left, **so we need to find the index of the leftmost `target` in the array**. Initially consider implementing this through the steps shown in the figure below.
 
-1. Perform a binary search, get an arbitrary index of `target`, denoted as $k$.
-2. Start from index $k$, and perform a linear search to the left until the leftmost `target` is found and return.
+1. Perform a binary search to find any `target`'s index, say $k$.
+2. Starting from index $k$, perform a linear search to the left until the leftmost `target` is found and return.
```

Comment:
"until the leftmost `target` is found and return." --> "until the leftmost target is found, then return the index."